mne/preprocessing/ica.py

```diff
@@ -1850,6 +1856,71 @@ def _find_sources(sources, target, score_func):
     return scores
 
 
+def _ica_explained_variance(ica, inst, normalize=False):
+    """Checks variance accounted for by each component in supplied data.
+
+    parameters
+    ----------
+    ica : ICA
+        Instance of `mne.preprocessing.ICA`.
+    inst : Raw | Epochs | Evoked
+        Data to explain with ICA. Instance of Raw, Epochs or Evoked.
+    normalize : bool
+        Whether to normalize the variance.
+
+    returns
+    -------
+    var : array
+        Variance explained by each component.
+    """
+    # check if ica is ICA and whether inst is Raw or Epochs
+    if not isinstance(ica, ICA):
+        raise TypeError('first argument must be an instance of ICA.')
+    if not isinstance(inst, (_BaseRaw, _BaseEpochs, Evoked)):
+        raise TypeError('second argument must an instance of either Raw, '
+                        'Epochs or Evoked.')
+
+    src = ica.get_sources(inst)
```

Comment:
Just echoing what @choldgraf suggested: `get_data` method working for Raw, Epochs, Evoked etc. would make life a little easier :) (see code below)